test/cache-test.c

```diff
@@ -0,0 +1,149 @@
+#include <limits.h>
+#include <string.h>
+#include "rimraf/rimraf.h"
+#include "minunit/minunit.h"
```

Comment:
I'm -1 on having two unit testing frameworks here. If we're going to move everything here, let's standardize on a few things:

- code style
- tests/test frameworks
- build tools
- ... etc